outbound/hmail.js

```diff
@@ -87,29 +87,36 @@ class HMailItem extends events.EventEmitter {
 
     read_todo () {
         const self = this;
-        const tl_reader = fs.createReadStream(self.path, {start: 0, end: 3});
+        const length_stream = fs.createReadStream(self.path, {start: 0, end: 3});
 
-        tl_reader.on('error', (err) => {
+        length_stream.on('error', (err) => {
             const errMsg = `Error reading queue file ${self.filename}: ${err}`;
             self.logerror(errMsg);
             self.temp_fail(errMsg);
         });
 
         let todo_len_raw = '';
-        tl_reader.on('data', (data) => { todo_len_raw += data; });
-        tl_reader.on('end', () => {
-            const todo_len = new Buffer(todo_len_raw).readUInt32BE(0);
-            const td_reader = fs.createReadStream(self.path, {encoding: 'utf8', start: 4, end: todo_len + 3});
+        length_stream.on('data', (data) => { todo_len_raw += data; });
+        length_stream.on('end', () => {
+            const todo_len = Buffer.from(todo_len_raw).readUInt32BE(0);
+            self.logdebug(`todo header length: ${todo_len}`);
+            const header_stream = fs.createReadStream(self.path, {encoding: 'utf8', start: 4, end: todo_len + 3});
             self.data_start = todo_len + 4;
 
             let todo_raw = '';
-            td_reader.on('data', (data) => { todo_raw += data; });
-            td_reader.on('end', () => {
+            header_stream.on('data', (data) => { todo_raw += data; });
+            header_stream.on('end', () => {
                 if (Buffer.byteLength(todo_raw) === todo_len) {
                     // we read everything
+                    todo_raw = todo_raw.trim()
```

Comment:
Is there any situation this might happen or is it just belt and braces?

Reply:
It removes the leading \n characters (in a backwards compatible way) added here:

```js
const todo_str = `\n${JSON.stringify(todo, exclude_from_json)}\n`
```

Reply:
Oh - I was thinking "this isn't required for JSON.parse to work" but it's really for the last character check more than anything.